yaml_specs/modulemd_stream_v2.yaml

```diff
@@ -1,50 +1,125 @@
+##############################################################################
+# Glossary:                                                                  #
+#                                                                            #
+# build system: The process by which a module is built and packaged. In many #
+# cases, this will be the Module Build Service tool, but this term is used   #
+# as a catch-all to describe any mechanism for producing a yum repository    #
+# containing modular content from input module metadata files.               #
+#                                                                            #
+#                                                                            #
+# == Attribute Types ==                                                      #
+#                                                                            #
+# MANDATORY: Attributes of this type must be filled in by the packager of    #
+# this module. They must also be preserved and provided in the output        #
+# metadata produced by the build system for inclusion into a repository.     #
+#                                                                            #
+# OPTIONAL: Attributes of this type may be provided by the packager of this  #
+# module, when appropriate. If they are provided, they must also be          #
+# preserved and provided in the output metadata produced by the build        #
+# system for inclusion into a repository.                                    #
+#                                                                            #
+# AUTOMATIC: Attributes of this type must be present in the the repository   #
+# metadata, but they may be left unspecified by the packager. In this case,  #
+# the build system is responsible for generating an appropriate value for    #
+# the attribute and including it in the repository metadata. If the packager #
+# specifies this attribute explicitly, it must be preserved and provided in  #
+# the output metadata for inclusion into a repository.                       #
+##############################################################################
```

Comment:
There are cases where the build system overrides supplied values; for instance, MBS has APIs that let you override the NSVC or the dependencies. I'm not sure if the language here isn't too strict.

While we normally want to it preserved, the abovementioned behavior shouldn't be considered broken or as violating the standard, I'd say.

Reply:
@contyk 
Does the following line sound right if added at the end of the glossary?

"The definitions above describe the expected behavior of the build system operating in its default configuration. It is permissible for the build system to override user-provided entries through non-default operating modes. If such changes are made, all items indicated as being required for the output repository must still be present."

Reply:
Yeah, that'd be fine.